civitai-downloader/civitai_downloader/download.py

```
@@ -0,0 +1,425 @@
+        # Create model directory structure
+        model_dir = path / f"model_{version.model_id}" / f"version_{version.id}"
+        model_dir.mkdir(parents=True, exist_ok=True)
+        
+        # Download each file
+        for i, file_info in enumerate(version.files):
+            file_path = model_dir / file_info.name
+            
+            # Create progress callback for this specific file
+            def file_progress_callback(progress: DownloadProgress) -> None:
+                if progress_callback:
+                    progress_callback(f"File {i+1}/{len(version.files)}", progress)
```

Ruff (0.12.2)

Function definition does not bind loop variable `i`

(B023)